packages/db-objects-file/lib/objects/objectsInMemServerRedis.js

```diff
@@ -796,11 +806,11 @@ class ObjectsInMemoryServer extends ObjectsInMemoryFileDB {
 
             this.server.listen(
                 settings.port || 9001,
-                (settings.host && settings.host !== 'localhost') ? settings.host : ((settings.host === 'localhost') ? '127.0.0.1' : undefined),
+                settings.host === 'localhost' ? '127.0.0.1' : settings.host ? settings.host : undefined,
                 callback
             );
         } catch (e) {
-            callback && callback(e);
+            callback(e);
```

Comment:
Sorry... why callback instead of Promise here?

Reply:
;-) because connection/listening callbacks are common in the nodejs world and the callback is also dorectly passed through to the net.Server.listen method?

In fact yes we can decide to have promises in "all own code". In fact the whole class with all one event handler have a lot of callbacks all over the place because "it is common that way".

So on one side "yes it is the only place with a callback while having one other place (destroy) which is now async" and we should have it "the same".

So in fact it would be now a strategic decision if we want to go with Promises everywhere ...

Reply:
Why not our maybe cb which can do both?

Reply:
So the callback is only used because the Node.js internals use one?

IMO we don't have to switch everywhere immediately, but while we're refactoring stuff we should move the new code to Promises. Makes it easier to use from outside.

Reply:
Also, Node.js is slowly moving its callback-API to Promises:
https://nodejs.org/api/fs.html#fs_fs_promises_api

Reply:
In fact this is used in exactly one place ... but yes Ok I will change to Promises to end the discussion :-))

Reply:
Just seems weird that the whole API is async, except for this method.